torch/quantization/_equalize.py

```diff
@@ -0,0 +1,120 @@
+import torch
+import copy
+
+def channel_range(input, axis=0):
+    ''' finds the range of weights associated with a specific channel
+
+    '''
+    size_of_tensor_dim = len(input.size())
+    mins = input
+    maxs = input
+
+    # reshape input to make specified axis the last axis in input
+    temp = list(range(size_of_tensor_dim))
+    temp[size_of_tensor_dim - 1] = axis
+    temp[axis] = size_of_tensor_dim - 1
+
+    mins = mins.permute(temp)
+    maxs = maxs.permute(temp)
+
+    # minimizing over all axises except the last axis
+    for i in range(size_of_tensor_dim - 1):
+        mins = torch.min(mins, 0)[0]
+        maxs = torch.max(maxs, 0)[0]
+
+    assert(mins.size()[0] == input.size()[axis])
+    return maxs - mins
+
+def cross_layer_equalization(module1, module2, output_axis=0, input_axis=1):
+    ''' Given two adjacent tensors', the weights are scaled such that
+    the ranges of the first tensors' output channel are equal to the
+    ranges of the second tensors' input channel
+
+    '''
+    _supported_types = [torch.nn.Conv2d, torch.nn.Linear]
+    if type(module1) not in _supported_types or type(module2) not in _supported_types:
+        raise ValueError("module type not supported:", type(module1), " ", type(module2))
+
+    if module1.weight.size()[output_axis] != module2.weight.size()[input_axis]:
+        raise TypeError("Incompatible tensors")
+
+    weight1 = module1.weight
+    weight2 = module2.weight
+    bias = module1.bias
+
+    weight1_range = channel_range(weight1, output_axis)
+    weight2_range = channel_range(weight2, input_axis)
+
+    # producing scaling factors to applied
+    weight2_range += 1e-9
+    scaling_factors = torch.sqrt(weight1_range / weight2_range)
+    r_scaling_factors = torch.reciprocal(scaling_factors)
+
+    bias = bias * r_scaling_factors
```

Comment:
Maybe calling them inverse_scaling_factors as opposed to r_scaling_factors for better readability